File: Api/src/permission/domain/repository/permission.go
package repository

import "gitlab.clirisgroup.com/internal/account-admin-tool/Api/src/permission/domain/entity"

type PermissionRepository interface {
	GetAll() ([]entity.Permission, error)
	Get(id int) (*entity.Permission, error)
	Save(*entity.AddPermissionDTO) error
	Update(permission *entity.UpdatePermissionDTO, id int) error
	Delete(id int) error
}

File: frontend/domain/ports/permission/permission.ts
import AddPermissionDTO from '../../dto/permission/addPermissionDTO';
import PermissionDTO from '../../dto/permission/permissionDTO';
import UpdatePermissionDTO from '../../dto/permission/updatePermission';
import Permission from '../../models/permission/permission';

export default interface PermissionRepository {
    getPermissions(): Promise<PermissionDTO[]>;
    getPermission(id: number): Promise<Permission>;
    updatePermission(id: number, data: UpdatePermissionDTO): Promise<UpdatePermissionDTO>;
    save(data: AddPermissionDTO): Promise<void>;
    delete(id: number): Promise<void>;
}
